### PR TITLE
Backport: Blame: fix some issues with the stream (#63865)

### PR DESCRIPTION
**Pull request by camdencheek**
Contributes to SRCH-738

Notably, this does not yet identify the root cause of SRCH-738, but it does identify and fix some confounding bugs. It's possible that these actually also _cause_ some of the issues in SRCH-738, but I wanted to at least push these to dotcom, where we can reproduce some of the weirdness. At the very least, it doesn't explain the auth errors being reported.

(cherry picked from commit d91fab39e287d57d00f35fb8952f5b7daabfa1d2)

## Test plan

CI

### CODE REVIEW

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/page.spec.ts [hunk 1]
```diff
@@ -248,9 +248,8 @@ test('history panel', async ({ page, sg }) => {
     await expect(page.getByText('Test commit')).toBeHidden()
 })
 
-test('file popover', async ({ page, sg }, testInfo) => {
-    // Test needs more time to teardown
-    test.setTimeout(testInfo.timeout * 3000)
+test('file popover', async ({ page, sg }) => {
+    test.slow()
```
Comment (camdencheek) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/page.spec.ts:252]: Unrelated conflict resolution

**Review by craigfurman — APPROVED** [review]

**Comment by bahrmichael** [comment]
@craigfurman @DaedalusG This PR now contains tests fixes from main and is ready to merge. Let me know if there's anything I need to do to have it merged.

**Comment by craigfurman** [comment]
@bahrmichael you should be able to merge it now that its approved. If that's not the case, please ping here again and we can merge it 🙂 

@DaedalusG I'm a bit lost after recent release/backport changes, do the release team actually have to approve backports into release branches? https://www.notion.so/sourcegraph/Sourcegraph-Backport-Tool-211d005248bb4b588b4fe8401381e71c implies no - developers can approve backport PRs made by the bot themselves and merge them, according to that. Manual cherry-picks like this of course need to be reviewed by any other org member, because that's just how github works, but AFAIK it doesn't _have_ to be the release team. Holler if I'm wrong!

Zooming out, it doesn't make sense IMO for the release team to review all backports - to do so would imply we have reviewer-level context on every change in the company - but this comment is a question about the process today rather than one about how we might want to change it.

**Comment by bahrmichael** [comment]
> @bahrmichael you should be able to merge it now that its approved. If that's not the case, please ping here again and we can merge it 🙂

@craigfurman GitHub doesn't allow me to merge, because "The base branch restricts merging to authorized users." Can you  hit the button?

**Comment by craigfurman** [comment]
Sorry for the delay @bahrmichael, TIL that is the shape of the merge protection rule. Ironically I can't see it, not being a repo admin 😂

**Pull request merged**